tests/model_registry/conftest.py

```diff
@@ -293,3 +302,194 @@ def model_registry_operator_pod(admin_client: DynamicClient) -> Pod:
+        )
+        pytest.fail(f"Unexpected error getting token for SA {sa_name}: {e}")
+
+
+# --- RBAC Fixtures (Using Context Managers, Function Scoped) ---
+
+
+@pytest.fixture(scope="function")
+def mr_access_role(
+    admin_client: DynamicClient,
+    model_registry_namespace: str,  # Existing fixture from main conftest
+    sa_namespace: Namespace,  # Used for unique naming
+) -> Generator[Role, None, None]:
+    """
+    Creates the MR Access Role using direct constructor parameters and a context manager.
+    """
+    role_name = f"registry-user-{MR_INSTANCE_NAME}-{sa_namespace.name[:8]}"
+    LOGGER.info(f"Defining Role: {role_name} in namespace {model_registry_namespace}")
+
+    # Define rules directly as required by the Role constructor's 'rules' parameter
+    role_rules: List[Dict[str, Any]] = [
+        {
+            "apiGroups": [""],  # Core API group
+            "resources": ["services"],  # As per last refinement for REST access
+            "resourceNames": [MR_INSTANCE_NAME],  # Grant access only to the specific MR service object
+            "verbs": ["get"],
+        }
+    ]
+
+    # Define labels, to be passed via **kwargs
+    role_labels = {
+        "app.kubernetes.io/component": "model-registry-test-rbac",
+        "test.opendatahub.io/namespace": sa_namespace.name,
+    }
+
+    LOGGER.info(f"Attempting to create Role: {role_name} with rules and labels.")
+    # Use context manager for creation and deletion
+    # Pass rules and labels directly
+    with Role(
+        client=admin_client,
+        name=role_name,
+        namespace=model_registry_namespace,
+        rules=role_rules,
+        label=role_labels,  # Pass labels via kwargs
+    ) as role:
+        try:
+            role.wait(timeout=60)  # Wait for role object to exist
+            LOGGER.info(f"Role {role.name} created successfully.")
+            yield role
+            LOGGER.info(f"Role {role.name} deletion initiated by context manager.")
+        except Exception as e:  # Catch other potential errors during Role instantiation or wait
+            LOGGER.error(f"Error during Role {role_name} creation or wait: {e}", exc_info=True)
+            pytest.fail(f"Failed during Role {role_name} creation: {e}")
+
+
+@pytest.fixture(scope="function")
+def mr_access_role_binding(
+    admin_client: DynamicClient,
+    model_registry_namespace: str,  # Existing fixture from main conftest
+    mr_access_role: Role,  # Depend on the role fixture to get its name
+    sa_namespace: Namespace,  # The namespace containing the test SA
+) -> Generator[RoleBinding, None, None]:
+    """
+    Creates the MR Access RoleBinding using direct constructor parameters and a context manager.
+    """
+    binding_name = f"{mr_access_role.name}-binding"  # Simplify name slightly, role name is already unique
+    role_name_ref = mr_access_role.name  # Get the actual name from the created Role object
```

Comment:
no need to save. Use directly.

Reply:
I reuse it also in the logs so I think it's cleaner defining it here at the top, same with role_name

Reply:
It is just a python object property. There is no point saving here.

Reply:
Are you talking about role_name_ref?

Reply:
Right. But won't hold this PR for this comment.